### PR TITLE
Cleanup of source code

### PR DESCRIPTION
**Pull request by keileg**
## Proposed changes
This is a major PR containing several smaller sets of changes that for practical reasons were joined. Main changes:
* Final (?) cleanup of the old test folders.
* Deprecated various functions, classes and methods.
* Introduced a new library of standard MixedDimensional grids.

As the various changes have undergone code review before being merged into this larger branch, we will not undertake a formal code review of this PR.

## Types of changes

What types of changes does this PR introduce to PorePy?
_Put an `x` in the boxes that apply._

- [ ] Minor change (e.g., dependency bumps, broken links).
- [ ] Bugfix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [x] Testing (contribution related to testing of existing or new functionality).
- [x] Documentation (contribution related to adding, improving, or fixing documentation).
- [x] Maintenance (e.g., improve logic and performance, remove obsolete code).
- [ ] Other:

## Checklist

_Put an `x` in the boxes that apply or explain briefly why the box is not relevant._

- [x] The documentation is up-to-date.
- [x] Static typing is included in the update.
- [x] This PR does not duplicate existing functionality.
- [x] The update is covered by the test suite (including tests added in the PR).
- [x] If new skipped tests have been introduced in this PR, `pytest` was run with the `--run-skipped` flag.

### CODE REVIEW

**Comment by keileg** [comment]
DO NOT MERGE until having run 'all tests'.

File: src/porepy/applications/md_grids/mdg_library.py [hunk 1]
```diff
@@ -21,8 +21,8 @@ def square_with_orthogonal_fractures(
     grid_type: Literal["simplex", "cartesian", "tensor_grid"],
     meshing_args: dict[str, float],
     fracture_indices: list[int],
-    fracture_endpoints: Optional[list[[pp.number, pp.number]]] = None,
-    size: Optional[pp.number] = 1,
```
Comment (IvarStefansson) [inline src/porepy/applications/md_grids/mdg_library.py]: I thought optional parameters should be explicitly marked as such. Is that wrong?

**Comment by keileg** [comment]
All tests now pass.

File: tests/numerics/fv/test_upwind.py [hunk 2]
```diff
@@ -551,7 +551,11 @@ def _assertion(self, lhs, rhs, theta, lhs_ref, rhs_ref, theta_ref):
     """Test the case of 2d domain with 1d fractures."""
 
     def test_2d_1d(self):
-        mdg, _ = pp.md_grids_2d.two_intersecting([2, 2], simplex=False)
+        mdg, _ = pp.mdg_library.square_with_orthogonal_fractures(
+            "cartesian",
+            {"cell_size_x": 0.5, "cell_size_y": 0.5},
```
Comment (IvarStefansson) [inline tests/numerics/fv/test_upwind.py:556]: This could also be "cell_size": 0.5

File: tests/viz/test_exporter.py [hunk 3]
```diff
@@ -210,8 +210,11 @@ def test_mdg(setup: ExporterTestSetup):
     """
 
     # Define grid
-    mdg, _ = pp.md_grids_2d.two_intersecting(
-        [4, 4], y_endpoints=[0.25, 0.75], simplex=False
+    mdg, _ = pp.mdg_library.square_with_orthogonal_fractures(
+        "cartesian",
+        meshing_args={"cell_size_x": 0.25, "cell_size_y": 0.25},
```
Comment (IvarStefansson) [inline tests/viz/test_exporter.py:215]: Again, I think "cell_size": 0.25 would be neater here and below.

**Review by IvarStefansson — APPROVED** [review]

File: src/porepy/applications/md_grids/mdg_library.py (hunk 1 above)
Comment (keileg) [reply src/porepy/applications/md_grids/mdg_library.py]: `Optional` means the value can be `None.` In this case, the default will be `1`; you would have to explicitly write `(..., size=None, ...)` to get the effect of the `Optional`, and this is not really usage we want to account for.

File: tests/numerics/fv/test_upwind.py (hunk 2 above)
Comment (keileg) [reply tests/numerics/fv/test_upwind.py:556]: Agree. Will change.

**Pull request merged**